src/main/java/edu/harvard/iq/dataverse/engine/command/impl/SubmitToArchiveCommand.java

```diff
@@ -0,0 +1,247 @@
+package edu.harvard.iq.dataverse.engine.command.impl;
+
+import edu.harvard.iq.dataverse.DOIDataCiteRegisterService;
+import edu.harvard.iq.dataverse.DataCitation;
+import edu.harvard.iq.dataverse.Dataset;
+import edu.harvard.iq.dataverse.DatasetVersion;
+import edu.harvard.iq.dataverse.DvObject;
+import edu.harvard.iq.dataverse.DatasetLock.Reason;
+import edu.harvard.iq.dataverse.authorization.Permission;
+import edu.harvard.iq.dataverse.authorization.users.ApiToken;
+import edu.harvard.iq.dataverse.authorization.users.AuthenticatedUser;
+import edu.harvard.iq.dataverse.engine.command.Command;
+import edu.harvard.iq.dataverse.engine.command.CommandContext;
+import edu.harvard.iq.dataverse.engine.command.DataverseRequest;
+import edu.harvard.iq.dataverse.engine.command.RequiredPermissions;
+import edu.harvard.iq.dataverse.engine.command.exception.CommandException;
+import edu.harvard.iq.dataverse.settings.SettingsServiceBean;
+import edu.harvard.iq.dataverse.util.bagit.BagGenerator;
+import edu.harvard.iq.dataverse.util.bagit.OREMap;
+import edu.harvard.iq.dataverse.workflow.step.Failure;
+import edu.harvard.iq.dataverse.workflow.step.WorkflowStepResult;
+
+import static edu.harvard.iq.dataverse.engine.command.CommandHelper.CH;
+
+import java.io.IOException;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.nio.charset.Charset;
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import org.apache.commons.codec.binary.Hex;
+import org.apache.commons.io.IOUtils;
+import org.duracloud.client.ContentStore;
+import org.duracloud.client.ContentStoreManager;
+import org.duracloud.client.ContentStoreManagerImpl;
+import org.duracloud.common.model.Credential;
+import org.duracloud.error.ContentStoreException;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+@RequiredPermissions(Permission.ArchiveDatasetVersion)
```

Comment:
I get the idea here behind this, but I'd like us to think a little more on permissions before adding a new one. As an example, we currently keep move dataverse and link dataverse as superuser only, because depending on what / how you move things it could lead to major re indexing. One idea we had was to have a new permission that is effectively a pseudo-superuser at a dataverse level (I call it that, because superuser means outside of the permission system), that would let you do the advanced things. Archive Dataset maybe falls under that?

So, I'd rather us maybe keep this superuser only for now until we can fully decide on what permission revamping we want (similarly, we're considering removing DeleteDatasetDraft, as we allow anyone who can edit to delete drafts). 

If you feel strongly about this, please let me/us know and we can maybe schedule a call to discuss. To be clear, I'm not saying this should permanently be superuser only, just for now, while we figure out things.

Reply:
@scolapasta - I think my expectation was that people may want to allow curators (or other role) to submit versions, particularly from the GUI rather than someone who already had API access). Since the UI option is not going into the base version now, I don't think there's that much to gain from the permission. I'll go ahead and remove it.

Reply:
Can you explain again, in this case, what you mean by the curator? In other words, why wouldn't checking if they have the publishDataset permission be enough?

Reply:
Just the curator role. So it would work to reuse the publishDataset permission / that would be better than superuser. (It may be that there are still use cases where separate permissions make sense, but just not restricting to superuser only should address QDR's interests for now.) So I'll remove the permission but switch to requiring the publishDataset permission rather than restricting to superuser.